internal/yggdrasil/yggdrasil.go

```diff
@@ -310,33 +317,66 @@ func (h *Handler) updateDeploymentStatuses(oldDeployments []v1alpha1.Deployment,
 	return deployments
 }
 
-func (h *Handler) toWorkloadList(logger logr.Logger, deployments []v1alpha1.EdgeDeployment) models.WorkloadList {
+func (h *Handler) toWorkloadList(ctx context.Context, logger logr.Logger, deployments []v1alpha1.EdgeDeployment, device *v1alpha1.EdgeDevice) (models.WorkloadList, error) {
 	list := models.WorkloadList{}
 	for _, deployment := range deployments {
 		if deployment.DeletionTimestamp != nil {
 			continue
 		}
-		podSpec, err := yaml.Marshal(deployment.Spec.Pod.Spec)
+		spec := deployment.Spec
+		podSpec, err := yaml.Marshal(spec.Pod.Spec)
 		if err != nil {
 			logger.Error(err, "cannot marshal pod specification", "deployment name", deployment.Name)
 			continue
 		}
 		var data *models.DataConfiguration
-		if deployment.Spec.Data != nil && len(deployment.Spec.Data.Paths) > 0 {
+		if spec.Data != nil && len(spec.Data.Paths) > 0 {
 			var paths []*models.DataPath
-			for _, path := range deployment.Spec.Data.Paths {
+			for _, path := range spec.Data.Paths {
 				paths = append(paths, &models.DataPath{Source: path.Source, Target: path.Target})
 			}
 			data = &models.DataConfiguration{Paths: paths}
 		}
+
 		workload := models.Workload{
 			Name:          deployment.Name,
 			Specification: string(podSpec),
 			Data:          data,
 		}
+		authFile, err := h.getAuthFile(ctx, spec.ImageRepositories, deployment.Namespace)
```

Comment:
I'm having second thoughts about the naming, ImageRepositories vs ImageRegistries.
Short quiz ;-) : The auth file should allow the pod to pull its image from the image _[complete the missing word]_.

In the context of images and k8s, it seems that the term `registry` is commonly used for it. https://kubernetes.io/docs/concepts/containers/images/

To the user it should be clear what is the semantic of repository, but if it is a synonym to the registry, we should pick the latter. WDYT ?

Reply:
It would be probably clearer. The downside is that the data model would have to be changed for consistency.
WDYT about changing that in this PR?

Reply:
+1